### ISSUE TITLE
OpenJ9 build for JDK 17 is missing

### ISSUE THREAD

**Issue by kriegaex** [actions/setup-java#239]
**Description:**
I know that with migration from Adopt OpenJDK to Eclipse Temurin, Eclipse had to agree with Oracle to no longer provide OpenJ9 builds. But IBM stepped in and provides production-ready OpenJ9 binaries. Both facts are documented [here](https://blog.adoptopenjdk.net/2021/08/goodbye-adoptopenjdk-hello-adoptium/). I do not know if _setup-java_ still downloads its OpenJ9 packages from the legacy Adopt OpenJDK website or directly from the IBM Semeru website. At the moment, it does not matter much, because even IBM does not offer JDK 17 downloads yet. I am not sure about the reason, maybe OpenJ9 does not support Java 17 yet. Probably you know better than I. I did not do any deep research here. I just want some feedback, explaining what to expect here.

**Justification:**
We can use this issue as a reminder or to-do list task to include Semeru OpenJ9 JDK 17, as soon as it is available.

**Are you willing to submit a PR?**
No. I am just a user contributing to other OSS projects who happened to notice this issue when trying to upgrade versions in the GitHub build matrix for same projects.

**Comment by bmarwell** [comment]
OpenJ9, built as IBM Semeru has three milestone builds for Java 17 (https://github.com/ibmruntimes/semeru17-binaries/releases/) but it does have GA releases for Java 8, 11 and 16!

* IBM resource: https://developer.ibm.com/languages/java/semeru-runtimes/downloads/
* GitHub resources are `ibmruntimes/semeruNN-binaries` (where NN = 8, 11, 16, 17).

**Comment by kriegaex** [comment]
Thanks @bmarwell for the update concerning the milestone builds.

> * GitHub resources are `ibmruntimes/semeruNN-binaries` (where NN = 8, 11, 16, 17).

What does that mean in layman's terms? Does it mean that I can somehow use the milestone builds (at least the most recent one) with the setup-java action? Or was that just a hint for the maintainers here about how to grab those resources when upgrading the action?

**Comment by bmarwell** [comment]
Yup, just a hint for the maintainers. As long as there is no adoptium-marketplace-api and no IBM-api, they can just use their own github releases API :)

**Comment by bmarwell** [comment]
It is released, please add it when convenient. Thanks! :)

**Comment by Donnadam** [comment]
Please cut me out of this thread. I have unsubscribed from GitHub months
ago and keep getting these emails daily.
***@***.***

**Comment by kriegaex** [comment]
Actually, for me this works since I noticed on 2021-12-10 that a Semeru OpenJ9 release for Java 17 has been available, see https://github.com/dev-aspectj/aspectj-maven-plugin/commit/29c39f9b55f26dc7f876f1c6b3a73e11d6facdbc. So I am closing this issue.

**Comment by bmarwell** [comment]
I might use this issue to create a PR to the readme then.
